src/sentry/tagstore/snuba/backend.py

```diff
@@ -177,31 +181,37 @@ def __get_tag_keys_for_projects(
         if keys is not None:
             filters['tags_key'] = keys
         aggregations = [
-            ['uniq', 'tags_value', 'values_seen'],
             ['count()', '', 'count']
         ]
+
+        if include_values_seen:
+            aggregations.append(['uniq', 'tags_value', 'values_seen'])
         conditions = [['tags_key', 'NOT IN', self.EXCLUDE_TAG_KEYS]]
 
-        # TODO should this be sorted by count() descending, rather than the
-        # number of unique values
         result = snuba.query(
             start, end, ['tags_key'], conditions, filters, aggregations,
-            limit=limit, orderby='-values_seen',
-            referrer='tagstore.__get_tag_keys', **kwargs
+            limit=limit, orderby='-count', referrer='tagstore.__get_tag_keys',
+            **kwargs
         )
 
         if group_id is None:
             ctor = TagKey
         else:
             ctor = functools.partial(GroupTagKey, group_id=group_id)
 
-        return set([
-            ctor(
-                key=key,
-                values_seen=data['values_seen'],
-                count=data['count'],
-            ) for key, data in six.iteritems(result) if data['values_seen']
-        ])
+        results = set()
+        for key, data in six.iteritems(result):
+            params = {'key': key}
+            if include_values_seen:
+                params['values_seen'] = data['values_seen']
+                params['count'] = data['count']
+            else:
+                # If only one aggregate is requested then data is just that raw
```

Comment:
this is ok and should work fine, although I think it is the second caller workaround for my overly clever implementation in `utils.snuba.nest_groups()` so I might take a shot at fixing that later so the callers don't have to be tricky with the result.